WeartGloves/src/WeArtGlove.cpp

```diff
@@ -0,0 +1,1598 @@
+    if (!(config.check("human_finger_list") && config.find("human_finger_list").isList()))
+    {
+        yError() << LogPrefix << "Unable to find human_finger_list in the config file.";
+        return false;
+    }
+    fingerListYarp = config.find("human_finger_list").asList();
+
+    for (size_t i = 0; i < fingerListYarp->size(); i++)
+    {
+        humanFingerNames.push_back(fingerListYarp->get(i).asString());
+    }
+    yInfo() << LogPrefix << "human finger names: " << humanFingerNames;
+
+    // Fill the coupling Matrix
+    yarp::os::Bottle* closureToHumanJointYarp;
+    if (trackingAlgorithmType == "SCTA")
+    {
+        if (!(config.check("closure_to_human_joint_simple")) || !config.find("closure_to_human_joint_simple").isList())
+        {
+            yError() << LogPrefix << "couldn't find the proper coupling matrix";
+            return false;
+        }
+        closureToHumanJointYarp = config.find("closure_to_human_joint_simple").asList();
+
+        if (closureToHumanJointYarp->size() < 60)
+        {
+            yError() << LogPrefix << "size of the coupling matrix for simple tracking is:" << closureToHumanJointYarp->size() << "Required 60";
+            return false;
+        }
+        closureToHumanJoint.resize(20, N_THIMBLES);
+        for (int j = 0; j < closureToHumanJointYarp->size(); j++)
+        {
+            int col = j % N_THIMBLES;
+            int row = j / N_THIMBLES;
+            closureToHumanJoint(row, col) = closureToHumanJointYarp->get(j).asFloat64();
+        }
+    }
+    else if (trackingAlgorithmType == "ASTA")
+    {
+        if (!(config.check("closure_to_human_joint_rawSensor")) || !config.find("closure_to_human_joint_rawSensor").isList())
+        {
+            yError() << LogPrefix << "couldn't find the proper coupling matrix";
+            return false;
+        }
+        closureToHumanJointYarp = config.find("closure_to_human_joint_rawSensor").asList();
+
+        if (closureToHumanJointYarp->size() < 260)
+        {
+            yError() << LogPrefix << "size of the coupling matrix for rawSensor tracking is:" << closureToHumanJointYarp->size() << "Required 260";
+            return false;
+        }
+        closureToHumanJoint.resize(20, 13);
+        for (size_t j = 0; j < closureToHumanJointYarp->size(); j++)
+        {
+            size_t col = j % consideredJointNameList.size();
+            size_t row = j / consideredJointNameList.size();
+
+            closureToHumanJoint(row, col) = closureToHumanJointYarp->get(j).asFloat64();
+        }
+        if (closureToHumanJoint.cols() != consideredJointNameList.size())
+        {
+            yError() << LogPrefix << "Sizes of the closureToHumanJoint and consideredJointNameList do not match." << closureToHumanJoint.cols();
+            return false;
+        }
+
+        // Get URDF model path
+        std::string modelFile = config.check("model", yarp::os::Value("XSenseTestThreeFingers.urdf")).asString();
```

Comment:
This requires editing the ``YARP_DATA_DIRS`` and should be properly documented

Reply:
Good point. I added the explanation to the README as well as the code.